libs/@local/graph/api/src/rpc/mod.rs

```diff
@@ -34,3 +56,47 @@ impl SubsystemIdentifier for GraphSubsystemId {
         }
     }
 }
+
+pub struct Dependencies<S, A, C> {
+    pub store: Arc<S>,
+    pub authorization_api: Arc<A>,
+    pub temporal_client: Option<TemporalClient>,
+    pub codec: C,
+}
+
+#[expect(
+    clippy::significant_drop_tightening,
+    reason = "false-positive in `AccountServer`"
+)]
+pub fn rpc_router<S, A, C, N>(
+    dependencies: Dependencies<S, A, C>,
+    notifications: N,
+) -> (
+    Router<impl Route<RequestBody, ResponseBody: Send, Future: Send> + Send>,
+    Task<Account, N>,
+)
+where
+    S: StorePool + Send + Sync + 'static,
+    A: AuthorizationApiPool + Send + Sync + 'static,
+    C: ReportEncoder + ReportDecoder + Clone + Send + Sync + 'static,
+{
+    let builder = RouterBuilder::new(dependencies.codec)
+        .with_builder(|builder| {
+            builder
+                .layer(HandleReportLayer::new())
+                .layer(HandleBodyReportLayer::new())
+        })
+        .register(AuthenticationDelegate::new(AuthenticationServer))
+        .register(AccountDelegate::new(AccountServer {
+            store_pool: dependencies.store,
+            authorization_api_pool: dependencies.authorization_api,
+            temporal_client: dependencies.temporal_client.map(Arc::new),
```

Comment:
I didn't know that the account system requires a temporal client, however, it also doesn't hurt to have it. Especially because we can use this code for other systems as well.

Reply:
The temporal client is required for `store_pool.acquire`, see: https://github.com/hashintel/hash/blob/03a1cebc399918262c2176fe80e7af9ed8d89e59/libs/@local/graph/api/src/rest/account.rs#L176-L177 and https://github.com/hashintel/hash/blob/03a1cebc399918262c2176fe80e7af9ed8d89e59/libs/@local/graph/api/src/rpc/account.rs#L236-L237

Reply:
Ah, yes, not very proud of this part, though :D (let's keep it as it is, that has nothing to do with this PR).